tools/EteriTools/Preprocessing/preprocessing.xml

```diff
@@ -0,0 +1,111 @@
+<tool id="preproc" name="Preprocessing" version="0.1">
+
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command>
+		<![CDATA[
+
+  perl '$__tool_directory__/preprocessing.pl'
+    '$fastaFile'  $max_length $in_winShift $min_seq_length
+
+]]>
+	</command>
+    <inputs>
+        <param type="data" name="fastaFile" format="fasta" />
+
+        <param name="max_length" type="integer" value="10000" size="5" label="window size">
+          <sanitizer>
+               <valid initial="string.printable">
+                  <remove value="&apos;"/>
+               </valid>
+          </sanitizer>
+        </param>
+
+        <param name="in_winShift" type="integer" value="100" size="5" label="window shift in percent">
+          <sanitizer>
+               <valid initial="string.printable">
+                  <remove value="&apos;"/>
+               </valid>
+          </sanitizer>
+        </param>
+
+        <param name="min_seq_length" type="integer" value="5" size="5" label="minimum sequence length">
+          <sanitizer>
+               <valid initial="string.printable">
+                  <remove value="&apos;"/>
+               </valid>
+          </sanitizer>
+        </param>
+    </inputs>
+
+    <outputs>
+
+        <data name="data.fasta" format="fasta" from_work_dir="FASTA/data.fasta" label="data.fasta"/>
+        <data name="data.map" format="txt" from_work_dir="FASTA/data.map" label="data.map"/>
+        <data name="data.names" format="txt" from_work_dir="FASTA/data.names" label="data.names"/>
+        <data name="data.fasta.scan" format="fasta" from_work_dir="FASTA/data.fasta.scan" label="data.fasta.scan"/>
+        <data name="FASTA" format="zip" from_work_dir="FASTA.zip" label="FASTA.ZIP"/>
+    </outputs>
+
+
+    <tests>
+    <test>
+        <param name="fastaFile" value="input.fa"/>
+        <param name="max_length" value="10000"/>
+        <param name="in_winShift" value="100"/>
+        <param name="min_seq_length" value="5"/>
+        <output name="data.fasta" file="FASTA/data.fasta"/>
+        <output name="data.map" file="FASTA/data.map" />
+        <output name="data.names" file="FASTA/data.names"/>
+        <output name="data.fasta.scan" file="FASTA/data.fasta.scan" />
+        <!--output name="data.lens" file="FASTA/data.lens" /-->
+        <!--output name="data.locations" file="FASTA/data.locations" /-->
+        <!--output name="data.fasta.all_frags" file="FASTA/data.fasta.all_frags" /-->
+        <!--output name="orig.fasta" file="FASTA/orig.fasta"/-->
+        <!--output name="fragments.splitMasked.list" file="FASTA/fragments.splitMasked.list" /-->
+        <!--output name="FASTA" file="FASTA.zip"/-->
```

Comment:
yes it's really needed, because otherwise i have to pass to many separate files to the tools. In some cases it's more useful like this.